packages/rum-core/src/domain/rumEventsCollection/action/trackActions.ts

```diff
@@ -65,142 +55,198 @@ export interface ActionContexts {
+  const { stop: stopListener } = listenClickEvents((event) => {
+    onClick(state, event)
   })
 
   const actionContexts: ActionContexts = {
     findActionId: (startTime?: RelativeTime) =>
       isExperimentalFeatureEnabled('frustration-signals')
-        ? history.findAll(startTime).map((controller) => controller.id)
-        : history.find(startTime)?.id,
+        ? state.history.findAll(startTime)
+        : state.history.find(startTime),
   }
 
   return {
     stop: () => {
-      history.findAll().forEach((actionController) => actionController.discard())
+      state.stopObservable.notify()
       stopListener()
     },
     actionContexts,
   }
 }
 
-function listenEvents(callback: (event: Event & { target: Element }) => void) {
+function listenClickEvents(callback: (clickEvent: MouseEvent & { target: Element }) => void) {
   return addEventListener(
     window,
     DOM_EVENT.CLICK,
-    (event) => {
-      if (event.target instanceof Element) {
-        callback(event as Event & { target: Element })
+    (clickEvent: MouseEvent) => {
+      if (clickEvent.target instanceof Element) {
+        callback(clickEvent as MouseEvent & { target: Element })
       }
     },
     { capture: true }
   )
 }
 
-function newAction(
-  lifeCycle: LifeCycle,
-  domMutationObservable: Observable<void>,
-  type: AutoActionType,
-  name: string,
-  event: Event,
-  onCompleteCallback: (endTime: TimeStamp) => void,
-  onDiscardCallback: () => void
-): ActionController {
-  const id = generateUUID()
+function onClick(state: TrackActionsState, event: MouseEvent & { target: Element }) {
+  if (!state.collectFrustrations && state.history.find()) {
+    // TODO: remove this in a future major version. To keep retrocompatibility, ignore any new
+    // action if another one is already occurring.
+    return
+  }
+
+  const name = getActionNameFromElement(event.target, state.actionNameAttribute)
+  if (!state.collectFrustrations && !name) {
+    // TODO: remove this in a future major version. To keep retrocompatibility, ignore any action
+    // with a blank name
+    return
+  }
+
   const startClocks = clocksNow()
-  const eventCountsSubscription = trackEventCounts(lifeCycle)
+
+  const singleClickPotentialAction = newPotentialAction(state, {
+    name,
+    event,
+    type: ActionType.CLICK as const,
+    startClocks,
+  })
+
   const { stop: stopWaitingIdlePage } = waitIdlePage(
-    lifeCycle,
-    domMutationObservable,
+    state.lifeCycle,
+    state.domMutationObservable,
     (idleEvent) => {
-      if (idleEvent.hadActivity && startClocks.timeStamp <= idleEvent.end) {
-        complete(idleEvent.end)
+      if (!idleEvent.hadActivity) {
+        // If it has no activity, consider it as a dead click.
+        // TODO: this will yield a lot of false positive. We'll need to refine it in the future.
+        if (state.collectFrustrations) {
+          singleClickPotentialAction.frustrations.add(FrustrationType.DEAD)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:**  To be a bit more expressive, what do you think of having something like `singleClickPotentialAction.markAsFrustrating(FrustrationType.DEAD)`?